nixos/gitlab: do not set default http(s) port number of gitlab registry explicitly

brussee opened the pull request:
The external port number is part of the external url of the gitlab registry. By setting `port = null;` the port won't be part of the external url.
For example:
(Before this change:) `registry.gitlab.example.com:443` and `registry.gitlab.example.com:80` will become (after this change:) `registry.gitlab.example.com` for HTTPS and HTTP respectively.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

brussee commented:
Improvement upon: #351496

Thanks in advance @globin @krav @talyz @yayayayaka

illdefined commented:
May I suggest that you check out the [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) in the contributing guidelines and amend both the commit message and the PR title accordingly?

brussee committed:
nixos/gitlab: do not set default http(s) port number of gitlab registry explicitly

The external port number is part of the external url of the gitlab registry. By setting `port = null;` the port won't be part of the external url.
For example:
(Before this change:) `registry.gitlab.example.com:443` and `registry.gitlab.example.com:80` will become (after this change:) `registry.gitlab.example.com` for HTTPS and HTTP respectively.

brussee commented:
> May I suggest that you check out the [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) in the contributing guidelines and amend both the commit message and the PR title accordingly?

Thank you for being so polite, very much appreciated!

I've updated the commit message and PR title following the conventions of nixpkgs & nixos modules.

Please let me know if there's anything else I should do. Thanks in advance!

brussee commented:
> Please let me know if this is ready to merge or needs further improvements.
> 
> Thanks in advance @globin @krav @talyz @yayayayaka

Friendly reminder of this open PR.
